app/controllers/katello/concerns/api/v2/registration_controller_extensions.rb

```diff
@@ -35,14 +35,26 @@ def context_urls
 
       def smart_proxy
         @smart_proxy ||= begin
-          proxy = params[:url] ? SmartProxy.unscoped.find_by(url: params[:url]) : SmartProxy.pulp_primary
+          proxy = params[:url] ? find_smart_proxy_by_url : SmartProxy.pulp_primary
 
           fail Foreman::Exception, _('Smart proxy content source not found!') unless proxy
           fail Foreman::Exception, _('Pulp 3 is not enabled on Smart proxy!') unless proxy.pulp3_enabled?
 
           proxy
         end
       end
+
+      # Find proxy by URL configured in templates module
+      # or by proxy's URL
+      def find_smart_proxy_by_url
+        proxy_from_feature = SmartProxyFeature.where(feature: Feature.find_by(name: 'Templates'))
+                                              .where('settings LIKE ?', "%#{params[:url]}%")
```

Comment:
https://www.binarywebpark.com/query-json-data-rails-postgresql/ suggests the native way to query JSON is:
```suggestion
                                              .where('settings ->> 'template_url' = ?', params[:url])
```